frontend/src/directives/RefreshableIframe.es6

```diff
@@ -0,0 +1,18 @@
+/* @ngInject */
+export default () => {
+    return {
+        restrict: 'A',
+        link: function (scope, element, attr) {
+            var refreshMe = function () {
+                element.attr('src', element.attr('refreshable-src'));
+            };
+
+            scope.$watch('refreshIframe', function (newVal, oldVal) {
```

Comment:
Since this directive does not use an isolated scope you can add the refreshMe function onto the scope and then call it directly from the view as opposed to going through a controller

so the save & update button would be like ng-click="ctrl.save(promotion, refreshIframe);" - Where the second argument to save is a callback to run when saving is finished